crates/net/downloaders/src/metrics.rs

```diff
@@ -3,27 +3,29 @@ use reth_interfaces::p2p::error::DownloadError;
 use reth_metrics_derive::Metrics;
 
 /// The header downloader metrics.
-#[derive(Metrics)]
-#[metrics(scope = "downloaders.headers")]
-pub struct HeaderDownloaderMetrics {
-    /// The number of headers that were successfully sent to the poller (stage)
+#[derive(Clone, Metrics)]
+#[metrics(dynamic = true)]
+pub struct DownloaderMetrics {
```

Comment:
this should document that it will be initialized with dynamic scope (headers/bodies) and is used by both downloaders otherwise could be confusing.